[PL-Doc] Update platform release notes 1.89x (#10292)
harness/developer-hub: release-notes/platform.md

## May 2025

### Version 1.89.x <!--May 13, 2025-->

#### Fixed issues

> - Fixed an issue where the value for a reference-type secret was missing during policy evaluation. With this fix, the value is now passed correctly, ensuring that policy evaluation is no longer skipped.[PL-62417]

Reviewer: ```suggestion
- Fixed an issue where the value for a reference-type secret was missing during policy evaluation. With this fix, the value is now passed correctly, ensuring that policy evaluation on values is honoured. [PL-62417]
```

> - Resolved a problem where deleted variables did not reflect correctly in Resource Groups. Now, if a variable is deleted, it will also be removed from any associated Resource Groups.[PL-60850]

Reviewer: ```suggestion
- Resolved an issue where deleted variables did not reflect correctly in Resource Groups. Now, if a variable is deleted, it will also be removed from any associated Resource Groups. [PL-60850]
```

> - Updated the **User Group APIV2** to align with actual API behavior. The `UserInfo` object sample response previously listed all attributes; it now correctly displays only the subset of fields returned by the API. [PL-62617] 

Reviewer: There is no API update. Instead, only the docs are updated to reflect the correct API response.

Author: Noted.

> - Updated the **User Group APIV2** to align with actual API behavior. The `UserInfo` object sample response previously listed all attributes; it now correctly displays only the subset of fields returned by the API. [PL-62617] 

Reviewer: ```suggestion
- Updated the API docs for [**User Group APIV2** API](https://apidocs.harness.io/tag/User-Group/#operation/getUserGroupV2) to correctly reflect the subset of fields returned in the response. [PL-62617] 
```
